### PR TITLE
Introducing vpc-go-sdk supporting IBM Cloud VPC service

### PR DESCRIPTION
**Pull request by astha-jain**

### CODE REVIEW

File: developer/examples/build.vpc.windows.pkr.hcl [hunk 1]
```diff
@@ -17,19 +17,39 @@ variable "ansible_inventory_file" {
   default = "${env("ANSIBLE_INVENTORY_FILE")}"
 }
 
+variable "subnet_id" {
```
Comment (juanpinzon) [inline developer/examples/build.vpc.windows.pkr.hcl:20]: Packer Template variables should be introduce as plain text on the template. I know it is reasonable to think about store them on the env file, however, there is no need to store them as environment variables. Environment variables meaning is to store sensitive data like API keys, or Ansible and Packer configuration variables due you cannot publish them on a public repo for security reasons. The main reason behind is we want to keep industry standards, so clients does not have any problem when they move from other cloud providers (i.e. AWS Packer plugin)

File: builder/ibmcloud/vpc/step_create_vpc_service_instance.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+type stepCreateVPCServiceInstance struct {
+}
+
+func (step *stepCreateVPCServiceInstance) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	client := state.Get("client").(*IBMCloudClient)
+	ui := state.Get("ui").(packer.Ui)
+	iamurl := os.Getenv("AUTH_URL")
```
Comment (juanpinzon) [inline builder/ibmcloud/vpc/step_create_vpc_service_instance.go:20]: Avoid input variables from other sources than config.go file. For instance this piece of code should looks like:
@ step_create_vpc_service_instance
```
	config := state.Get("config").(Config)

	ui.Say("Creating VPC service...")

	options := &vpcv1.VpcV1Options{
		Authenticator: &core.IamAuthenticator{
			ApiKey: client.IBMApiKey,
			URL:    config.EndPointIAM,
		},
		URL: config.EndPoint,
	}
```
	
@ config.go
```
type Config struct {
        ....
	EndPointIAM      string `mapstructure-to-hcl2:",skip"`
	EndPoint         string `mapstructure-to-hcl2:",skip"`
	....
	if c.Region == "" {
		errs = packer.MultiErrorAppend(errs, errors.New("a region must be specified"))
	}
       // Configure IBM Cloud EndPoint and EndPointIAM
	c.EndPointIAM = "https://iam.test.cloud.ibm.com/identity/token/"
	// c.EndPoint = "https://" + c.Region + "-stage01.iaasdev.cloud.ibm.com/v1/"
	c.EndPoint = "https://" + c.Region + ".iaas.cloud.ibm.com/v1/"	
```
	
Also note that at this point the EndPoint MUST be production not testing

**Review by juanpinzon — CHANGES_REQUESTED** [review]
Hi Astha, I have been reviewing your PR. Here are some findings:
Also, don't forget before you commit any changes to hcl files, it is recommended to format them using packer. Example:
```
packer fmt examples/.
packer fmt developer/examples/.
```

File: builder/ibmcloud/vpc/step_create_vpc_service_instance.go (hunk 2 above)
Comment (astha-jain) [reply builder/ibmcloud/vpc/step_create_vpc_service_instance.go]: Addressed

**Comment by juanpinzon** [comment]
Could you please fix DCO issues.

**Pull request merged**